texttospeech/ssml_addresses/tts.py

```diff
@@ -0,0 +1,135 @@
+
+
+# [START tts_ssml_address_audio]
+def ssml_to_audio(ssml_text, outfile):
+    # Generates SSML text from plaintext.
+
+    # Given a string of SSML text and an output file name, this function
+    # calls the Text-to-Speech API. The API returns a synthetic audio
+    # version of the text, formatted according to the SSML commands. This
+    # function stores the synthetic audio to the designated output file.
+
+    # Args:
+    # ssml_text: string of SSML text
+    # outfile: string name of file under which to save audio output
+
+    # Returns:
+    # nothing
+
+    # Checks to make sure ssml_text is not empty
+    if ssml_text is None:
+        print('ssml_to_audio() error. The input text does not exist.')
+        return
+
+    # Instantiates a client
+    client = texttospeech.TextToSpeechClient()
+
+    # Sets the text input to be synthesized
+    synthesis_input = texttospeech.types.SynthesisInput(ssml=ssml_text)
+
+    # Builds the voice request, selects the language code ("en-US") and
+    # the ssml voice gender ("MALE")
+    voice = texttospeech.types.VoiceSelectionParams(
+        language_code='en-US',
+        ssml_gender=texttospeech.enums.SsmlVoiceGender.MALE)
+
+    # Selects the type of audio file to return
+    audio_config = texttospeech.types.AudioConfig(
+        audio_encoding=texttospeech.enums.AudioEncoding.MP3)
+
+    # Performs the text-to-speech request on the text input with the selected
+    # voice parameters and audio file type
+    response = client.synthesize_speech(synthesis_input, voice, audio_config)
+
+    # Writes the synthetic audio to the output file.
+    with open(outfile, 'wb') as out:
+        out.write(response.audio_content)
+        print('Audio content written to file ' + outfile)
+    # [END tts_ssml_address_audio]
+
+
+# [START tts_ssml_address_ssml]
+def text_to_ssml(inputfile):
+    # Generates SSML text from plaintext.
+    # Given an input filename, this function converts the contents of the text
+    # file into a string of formatted SSML text. This function formats the SSML
+    # string so that, when synthesized, the synthetic audio will pause for two
+    # seconds between each line of the text file. This function also handles
+    # special text characters which might interfere with SSML commands.
+
+    # Args:
+    # inputfile: string name of plaintext file
+
+    # Returns:
+    # A string of SSML text based on plaintext input
+
+    # Parse lines of input file
+    try:
+        open_file = open(inputfile, 'r')
+        raw_lines = open_file.readlines()
+
+    # Checks to make sure that the input file exists
+    except IOError:
+        print('text_to_ssml() error. File ' + inputfile + ' does not exist.')
+        return
+
+    # Define SSML timed break between addresses
+    time = '"2s"'
+    brk = '<break time=' + time + '/>'
+
+    # Initialize SSML script
+    ssml = '<speak>'
+
+    # Iterate through lines of file
+    for line in raw_lines:
```

Comment:
Can probably just convert this for loop to:

```
break_ = '<break time=2s/>'
raw_lines = ''.join(raw_lines)
ssml =  '<speak>{}</speak>'.format(raw_lines.replace('\n', break_))
```

Reply:
I've split up the break time variable like that because of SSML weirdness with double quotes inside of single quotes. I'm going to leave as is to accommodate the cgi.escape that we decided to keep. Let me know if I should do differently!